python/ray/data/tests/test_dataset.py

```diff
@@ -3729,6 +3729,15 @@ def get_node_id():
         ctx.use_push_based_shuffle = original
 
 
+def test_random_block_order(ray_start_regular_shared):
+    ds = ray.data.range(12).repartition(4)
+    ds = ds.randomize_block_order(seed=0)
```

Comment:
Test both lazy and non lazy cases?

Reply:
Updated the test to include the lazy path. LMK if it makes sense to combine the cases by just `parametrize`ing `optimize_fuse_read_stages`. For now I chose not to because I think the current code is more obvious for the non-lazy case.